merkle/sparse_merkle_tree.go

```diff
@@ -192,7 +191,7 @@ func nodeIDFromAddress(size int, prefix []byte, index *big.Int, depth int) stora
 	if depth > len(ib)*8 {
 		copy(t[len(prefix)+depthBytes-len(ib):], ib)
 	} else {
-		copy(t[len(prefix):], ib)
+		copy(t[size-len(ib):], ib)
```

Comment:
Is there an easy explanation of what these two branches mean / do that could be comments?